### ISSUE TITLE
feat: MLX (Apple Silicon) backend support for 4B DiT models

### ISSUE THREAD

**Issue by ChuxiJ** [ace-step/ACE-Step-1.5#995]
## Summary

The MLX backend for Apple Silicon has a **critical bug** that will cause runtime failures with 4B DiT models. The `condition_embedder` layer in the MLX DiT model hardcodes input dimension equal to decoder hidden_size, but 4B models have a separate (smaller) encoder hidden_size.

### condition_embedder Dimension Mismatch

**File:** `acestep/models/mlx/dit_model.py`

**Line 473**: The condition_embedder is created with matching input/output dimensions:
```python
self.condition_embedder = nn.Linear(inner_dim, inner_dim, bias=True)
```
Where `inner_dim = hidden_size` (decoder hidden size).

**For 4B models:**
- Decoder hidden_size = 2560 → `inner_dim = 2560`
- Encoder hidden_size = 2048 → encoder outputs tensors of dim 2048
- At **line 566**: `encoder_hidden_states = self.condition_embedder(encoder_hidden_states)` 
- **CRASH**: Input shape `[B, L, 2048]` fed to `Linear(2560, 2560)` → dimension mismatch error

**PyTorch reference (correct pattern)** in `modeling_acestep_v15_turbo.py` line 1280:
```python
condition_dim = getattr(config, "encoder_hidden_size", None) or config.hidden_size
self.condition_embedder = nn.Linear(condition_dim, inner_dim, bias=True)
```

### 1. `acestep/models/mlx/dit_model.py` — Constructor (lines 424-441)

Add `encoder_hidden_size` parameter:
```python
def __init__(
    self,
    hidden_size: int = 2048,
    intermediate_size: int = 6144,
    num_hidden_layers: int = 24,
    num_attention_heads: int = 16,
    num_key_value_heads: int = 8,
    head_dim: int = 128,
    encoder_hidden_size: int = None,  # NEW: separate encoder dim for 4B
    ...
):
    inner_dim = hidden_size
    condition_dim = encoder_hidden_size or hidden_size  # NEW
    self.condition_embedder = nn.Linear(condition_dim, inner_dim, bias=True)  # FIX line 473
```

### 2. `acestep/models/mlx/dit_model.py` — from_config() (lines 610-629)

Pass `encoder_hidden_size` from config:
```python
@classmethod
def from_config(cls, config) -> "MLXDiTDecoder":
    return cls(
        hidden_size=config.hidden_size,
        ...
        encoder_hidden_size=getattr(config, "encoder_hidden_size", None),  # NEW
    )
```

### 3. `acestep/models/mlx/dit_convert.py` — Weight Conversion (lines 11-66)

The weight conversion is dimension-agnostic (copies weights as-is), so it should work **IF** the constructor creates layers with correct dimensions. No changes needed in conversion logic itself, but verify that:
- [ ] `condition_embedder.weight` shape matches: `[inner_dim, condition_dim]` = `[2560, 2048]` for 4B
- [ ] Weight loading doesn't silently reshape or truncate

### 4. `acestep/core/generation/handler/mlx_dit_init.py` — Init Handler (lines 9-43)

```python
mlx_decoder = MLXDiTDecoder.from_config(self.config)  # Line 28
```

This passes `self.config` which should contain `encoder_hidden_size` if it's a 4B model. **No code change needed** after fixing `from_config()`.

### 5. `acestep/models/mlx/dit_generate.py` — Generation Loop

**Line 566**: `encoder_hidden_states = self.condition_embedder(encoder_hidden_states)`

This will work correctly after fixing the constructor — no change needed in generation code.

### 6. MPS Memory Considerations

**File:** `acestep/core/generation/handler/init_service_memory_basic.py`

Apple Silicon unified memory is shared between CPU and GPU. For 4B DiT (~9GB weights), this impacts:
- Macs with 16GB RAM: Very tight — may need aggressive chunking
- Macs with 32GB RAM: Workable with offload
- Macs with 64GB+ RAM: Comfortable

- [ ] Review MPS memory thresholds (currently 32GB threshold at line 124)
- [ ] Test 4B on M1/M2/M3/M4 with various RAM configs

### 7. Default Values in Constructor

**Lines 426-431**: Default values only match 2B:
```python
hidden_size: int = 2048,      # 4B uses 2560
intermediate_size: int = 6144, # 4B uses 9728
num_hidden_layers: int = 24,   # 4B uses 32
num_attention_heads: int = 16,  # 4B uses 32
```

These defaults are only used if `from_config()` is bypassed. Low risk but worth documenting.

## Acceptance Criteria

- [ ] **CRITICAL**: `condition_embedder` uses `encoder_hidden_size` when available
- [ ] `from_config()` passes `encoder_hidden_size` to constructor
- [ ] 4B model loads and generates audio on Apple Silicon without crashes
- [ ] Weight conversion preserves correct dimensions for `condition_embedder`
- [ ] 2B models continue to work (backward compatible — defaults to `hidden_size` when `encoder_hidden_size` is None)
- [ ] MPS memory management works for 4B model sizes
